db/batchlog_manager.cc

```diff
@@ -217,10 +217,10 @@ future<> db::batchlog_manager::replay_all_failed_batches(post_replay_cleanup cle
                 mutations.emplace_back(fm->to_mutation(s));
             }
             return mutations;
-        }).then([this, limiter, written_at, size, fms] (utils::chunked_vector<mutation> mutations) {
-            if (mutations.empty()) {
-                return make_ready_future<>();
-            }
+        });
+
+        try {
+          if (!mutations.empty()) {
```

Comment:
Replace this with an early return:
```
if (mutations.empty()) {
   co_return stop_iteration::no;
}
```
And you avoid an extra scope for the rest of this function.